src/client/components/ShoppingCart/RemoveFromShoppingCart.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './ShoppingCart.styles.css';
+
+function RemoveFromShoppingCart(props) {
+  const { product, shoppingCart, setShoppingCart } = props;
+
+  const removeProductFromShoppingCart = () => {
+    
+    setShoppingCart((prev) => prev.filter((item) => item.id !== product.id));
+    const result = shoppingCart.find(({ id }) => id === product.id);
+    if (result === undefined) {
+      // console.log('product does not exist in cart');
+    } else {
+      // console.log('remove product from the cart');
+      const index = shoppingCart.indexOf(result);
+      if (index > -1) {
+        shoppingCart.splice(index, 1);
+        
+      }
+    }
+  };
+
+  return (
+    <div className="add_cart_button">
+      <div>
+        <button
+          className="close-button"
+          onClick={removeProductFromShoppingCart}
+          type="button"
+        >
+          X
+        </button>
+      </div>
+      <div className="discounted_price">
+        <div
+          style={{
+            display: product.discount !== 0 ? 'inline-block' : 'none',
+            textDecoration: 'line-through',
+            color: 'black',
+          }}
+        >
+          {product.currency} {product.price}
+        </div>
+        <div
+          style={{
+            display: product.discount === 0 ? 'inline-block' : 'none',
+            color: 'black',
+          }}
+        >
+          {product.currency} {product.price}
+        </div>
+        <div
+          style={{
+            display: product.discount !== 0 ? 'inline-block' : 'none',
+            color: 'black',
+          }}
+        >
+          &nbsp;&nbsp;{product.currency}{' '}
```

Comment:
Using HTML characters for visual presentation is a no-no. Instead apply styling to achieve what you want. For example:
```suggestion
          <span className="currency-field">{product.currency}</span>
```

And in CSS:

```
.currency-field {
  margin-left: 2em;
  margin-right: 2em;
}
```

(I don't know the actual values 2em is probably too big)

Reply:
thats a good suggestion.Done with that